File: tests/fixtures/sample-python-project-il/code/app/services/features.py
import json
from pathlib import Path

# Spec forbids FEATURE_EXPORT_V2 from any shipped config — the data-export
# feature is GA, so the flag must be removed. It still ships in
# config/features.json.
# IL-DRIFT: ForbiddenArtifact:feature-experimental-export / forbidden.feature-flag.FEATURE_EXPORT_V2.present


def load_flags() -> dict:
    path = Path(__file__).resolve().parents[2] / "config" / "features.json"
    return json.loads(path.read_text())

File: tests/fixtures/sample-csharp-project-il/code/Services/Features.cs
// Spec forbids FEATURE_EXPORT_V2 from any shipped config — the data-export
// feature is GA, so the flag must be removed. It still ships in
// config/features.json.
// IL-DRIFT: ForbiddenArtifact:feature-experimental-export / forbidden.feature-flag.FEATURE_EXPORT_V2.present
using System.Text.Json;

namespace SampleApi.Services;

public static class Features
{
    public static Dictionary<string, bool> LoadFlags()
    {
        var path = Path.Combine(AppContext.BaseDirectory, "config", "features.json");
        var json = File.ReadAllText(path);
        return JsonSerializer.Deserialize<Dictionary<string, bool>>(json) ?? new();
    }
}
